luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -62,233 +29,462 @@ if gadgetHandler:IsSyncedCode() then
         {threshold = 475000, techLevel = 3.5},
         {threshold = 600000, techLevel = 4},
         {threshold = 725000, techLevel = 4.5}
-    }
+    },
+
+
+    --stores the values of powers for teams
+    PeakTeamPowers = {},
+    TeamPowers = {},
+
+    --stores the power of each living unit as a table {unitID, power, unitTeam}
+    UnitsWithPower = {},
+
+    
+    --teams
```

Comment:
Delete this comment

Reply:
To understand your methodology, if code explains itself comments aren't necessary. If comments aren't necessary, the code should probably be rewritten to explain itself.

Reply:
Either way, did as you asked.